test/integration/builder/main.go

```diff
@@ -109,8 +102,34 @@ func createOpenAPIBuilderConfig() *common.Config {
 // for testing.
 func createWebServices() []*restful.WebService {
 	w := new(restful.WebService)
-	// Define a dummy GET /test endpoint
-	w = w.Route(w.GET("test").
-		To(func(*restful.Request, *restful.Response) {}))
+	w.Route(buildRouteForType(w, "dummytype", "Foo"))
+	w.Route(buildRouteForType(w, "dummytype", "Bar"))
+	w.Route(buildRouteForType(w, "dummytype", "Baz"))
+	w.Route(buildRouteForType(w, "dummytype", "Waldo"))
+	w.Route(buildRouteForType(w, "listtype", "AtomicList"))
+	w.Route(buildRouteForType(w, "listtype", "MapList"))
+	w.Route(buildRouteForType(w, "listtype", "SetList"))
 	return []*restful.WebService{w}
 }
+
+// Implements OpenAPICanonicalTypeNamer
+var _ = util.OpenAPICanonicalTypeNamer(&typeNamer{})
+
+type typeNamer struct {
+	pkg  string
+	name string
+}
+
+func (t *typeNamer) OpenAPICanonicalTypeName() string {
+	return fmt.Sprintf("./testdata/%s.%s", t.pkg, t.name)
```

Comment:
I'm amused this doesn't follow the documented example return values... what's with the non-absolute path?

Reply:
type-based naming [enforces absolute path](https://github.com/golang/go/blob/ed15e82413c7b16e21a493f5a647f68b46e965ee/src/reflect/type.go#L76-L81)
namer-based naming doesn't. here we just document it

on the other hand, gengo supports [both absolute and relative path](https://github.com/kubernetes/gengo/blob/0689ccc1d7d65d9dd1bedcc3b0b1ed7df91ba266/parser/parse.go#L210-L212). `"./testdata/"` worked because we used to generate the [definition](https://github.com/kubernetes/kube-openapi/blob/d8b685cd2daadc40121e864b5e87ca8ededd9eed/test/integration/pkg/generated/openapi_generated.go#L32-L39) using relative path

switched to use absolute path instead